.github/workflows/integration.yaml

```diff
@@ -1,23 +1,35 @@
 name: Continuous Integration
 
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - master
+  pull_request:
 
 jobs:
-  unit-test:
-    name: Unit Test
+  lint:
+    name: Lint Code
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        charm: [pilot, ingressgateway]
+        charm: [pilot, gateway]
+    steps:
+    - uses: actions/checkout@v2
+    - run: sudo apt install tox
+    - run: tox -e ${{ matrix.charm }}-lint
 
+  unit:
+    name: Unit Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        charm: [pilot, gateway]
     steps:
     - uses: actions/checkout@v2
     - run: sudo apt install tox
-    - run: tox -e ${{ matrix.charm }}
+    - run: tox -e ${{ matrix.charm }}-unit
 
-  integration-test:
+  integration:
     name: Integration Test
```

Comment:
With this going to be a cross team effort I would like this to use pytest-operator.